### ISSUE TITLE
[util] test failure with Java 9

### ISSUE THREAD

**Issue by olebole** [Starlink/starjava#22]
Dear Mark,

we are currently switching to Java 9 on Debian, and found a problem in a `util` subpackage test:

```
Testcase: testResourceStream took 0.017 sec
FAILED
null
junit.framework.AssertionFailedError
  at junit.framework.Assert.fail(Assert.java:47)
  at junit.framework.Assert.assertTrue(Assert.java:20)
  at junit.framework.Assert.assertTrue(Assert.java:27)
  at uk.ac.starlink.util.DataSourceTest.testResourceStream(DataSourceTest.java:118)
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.base/java.lang.reflect.Method.invoke(Method.java:564)
  at junit.framework.TestCase.runTest(TestCase.java:164)
  at junit.framework.TestCase.runBare(TestCase.java:130)
  at junit.framework.TestResult$1.protect(TestResult.java:106)
  at junit.framework.TestResult.runProtected(TestResult.java:124)
  at junit.framework.TestResult.run(TestResult.java:109)
  at junit.framework.TestCase.run(TestCase.java:120)
  at junit.framework.TestSuite.runTest(TestSuite.java:230)
  at junit.framework.TestSuite.run(TestSuite.java:225)
  at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:536)
  at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:1205)
  at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:1048)
```
This happens when one tries to build this package, full build log [here](https://tests.reproducible-builds.org/debian/rbuild/unstable/amd64/starjava-util_1.0+2017.09.28-1.rbuild.log). I git it reported as Debian bug [#893529](https://bugs.debian.org/893529).

**Comment by olebole** [comment]
The erroring [lines](https://github.com/Starlink/starjava/blob/e66def1ae2cd49a083319c62e714191746c91e5b/util/src/testcases/uk/ac/starlink/util/DataSourceTest.java#L119-L120) are

```
        assertTrue( rSrc.getURL().toString()
                        .endsWith( "!/java/lang/Object.class" ) );
```

Printing out `rSrc.getURL().toString()` gives `jrt:/java.base/java/lang/Object.class`; so removing the exclamation mark solves the problem.

**Comment by mbtaylor** [comment]
Fix as suggested at e9fb7204cb4c60.  Thanks for the report.
